### PR TITLE
chore: bump abitype

### PR DESCRIPTION
**Pull request by tmm**
- Bumps [ABIType](https://abitype.dev)
- Removes custom config
- Refactors to `AbiFunction` and `AbiConstructor`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -133,6 +132,7 @@
     "@vitest/coverage-c8": "^0.29.2",
     "@vitest/ui": "^0.29.2",
     "@wagmi/cli": "^0.1.6",
+    "abitype": "^0.7.1",
```
Comment (tmm) [inline package.json:135]: I'm good locking it, but can add `~` back if you want.

File: src/types/abitype.ts [hunk 2]
```diff
@@ -1,14 +1,3 @@
-import type { Hex } from './misc'
-
-declare module 'abitype' {
-  export interface Config {
-    BytesType: {
-      inputs: Hex
-      outputs: Hex
-    }
-  }
-}
```
Comment (tmm) [inline src/types/abitype.ts:10]: This is the new default behavior in `abitype@0.7`

File: src/types/abitype.ts [hunk 3]
```diff
@@ -1,14 +1,3 @@
...
-
 export type {
   Abi,
   AbiError,
```
Comment (tmm) [inline src/types/abitype.ts:3]: Can we remove this file? Seems easier to import these from abitype directly rather than an internal export.

Comment (jxom) [reply src/types/abitype.ts:3]: Go for it

**Review by jxom — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (jxom) [reply package.json:135]: Doesn't abitype need to be a dependency?

**Pull request merged**